backend/src/server/context.ts

```diff
@@ -14,4 +29,9 @@ const context = (args: any) => {
   return context
 }
 
+export const getUser = (context: Context): dbUser => {
+  if (context.user) return context.user
+  throw new Error('No user given in context!')
+}
```

Comment:
cant we make this function a part of context?

`const user = context.getUser()`

This will automatically throw if there is no user and make those checks unnecessary `if (moderatorUser && moderatorUser.id....`

Furthermore  this function could be extended that it can actually query the user from the database when needed and serve as cache:

```
getUser(){
  if user return user
  if jwt query and return user
  throw no user
}
```

Reply:
An advance validation mechanic would be:

```
@MiddelWare(requiresuserMiddelware)
myQuery(args, UserProvidedByMidelWare: dbUser){
use user....
}
```
But I guess this would be an bigger exploration to find out if this works and is useful.
The advantage of this mechanic is, that a call would need to state if it needs the user and if so it would fail when its not presented. Furthermore the code inside the call would not to handle error cases etc since thats handled in the middleware already. The compiler would be tasked with throwing errors when the user is required in the call's code, but not acquired via middleware.

As I said - a bit far fetched - I like your solution - if you don't mind check my remarks and we are good to go on this.